sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/TaskUtils.java

```diff
@@ -531,7 +531,9 @@ public static boolean needsRecovery(TaskSpec taskSpec, TaskStatus taskStatus) {
         // Tasks with a goal state of finished should never leave the purview of their original
         // plan, so they are not the responsibility of recovery.  Recovery only applies to Tasks
         // which reached their goal state of RUNNING and then later failed.
-        if (taskSpec.getGoal().equals(GoalState.ONCE) || taskSpec.getGoal().equals(GoalState.FINISH)) {
```

Comment:
Could have a switch statement that explicitly lists the goal states. IDEs/linters can then catch if a goal state is created and then isn't added here.